### PR TITLE
mm256_srli,slli_si256; mm256_bsrli,bslli_epi128 to const generics

### PR DESCRIPTION
**Pull request by minybot**
f16c: _mm256_cvtps_ph; mm_cvtps_ph

### CODE REVIEW

**Comment by lqd** [comment]
I'm still wondering about the fact that we "* 8" the immediates that are supposed to be in bytes and <= 16 for the shifts

**Comment by Amanieu** [comment]
Yes, see #1012.

File: crates/core_arch/src/x86/f16c.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 use crate::{
     core_arch::{simd::*, x86::*},
-    hint::unreachable_unchecked,
+    //    hint::unreachable_unchecked,
```
Comment (Amanieu) [inline crates/core_arch/src/x86/f16c.rs:7]: Deleted commented code.

File: crates/core_arch/src/x86/avx2.rs [hunk 2]
```diff
@@ -2565,35 +2565,67 @@ pub unsafe fn _mm256_slli_epi64<const IMM8: i32>(a: __m256i) -> __m256i {
...
+    static_assert_imm8!(IMM8);
     let a = a.as_i64x4();
-    macro_rules! call {
-        ($imm8:expr) => {
-            vpslldq(a, $imm8)
-        };
-    }
-    transmute(constify_imm8!(imm8 * 8, call))
+    let r = vpslldq(a, IMM8 * 8);
+    transmute(r)
```
Comment (Amanieu) [inline crates/core_arch/src/x86/avx2.rs:2575]: You can just call `_mm256_bslli_epi128` here.

File: crates/core_arch/src/x86/avx2.rs [hunk 3]
```diff
@@ -2729,35 +2761,66 @@ pub unsafe fn _mm256_srav_epi32(a: __m256i, count: __m256i) -> __m256i {
...
+pub unsafe fn _mm256_srli_si256<const IMM8: i32>(a: __m256i) -> __m256i {
     let a = a.as_i64x4();
-    macro_rules! call {
-        ($imm8:expr) => {
-            vpsrldq(a, $imm8)
-        };
-    }
-    transmute(constify_imm8!(imm8 * 8, call))
+    let r = vpsrldq(a, IMM8 * 8);
+    transmute(r)
```
Comment (Amanieu) [inline crates/core_arch/src/x86/avx2.rs:2770]: You can just call `_mm256_bsrli_epi128` here.

**Pull request merged**